src/core/src/package_managers/YumPackageManager.py

```diff
@@ -347,26 +386,249 @@ def get_package_size(self, output):
     # region auto OS updates
     def get_current_auto_os_patch_state(self):
         """ Gets the current auto OS update patch state on the machine """
-        # NOTE: Implementation pending
-        pass
+        self.composite_logger.log("Fetching the current automatic OS patch state on the machine...")
+
+        # This function can be called for a specific auto update service (when called from disable auto OS updates code flow), or for all auto OS update services (when called from ConfigurePatchingProcessor to fetch status)
+        # apply_updates_identifier_text will be initialized only when this function is called for a specific auto update service. If not initialized, we need to check status from all services
+        if self.apply_updates_identifier_text == "":
+            for os_auto_update_service in self.auto_os_update_service_list:
+                if os_auto_update_service == self.yum_cron:
+                    self.__init_auto_update_for_yum_cron()
+                    self.__get_current_auto_os_updates_setting_on_machine()
+                    if self.enable_on_reboot_value or self.apply_updates_value == 'yes':
+                        break
+                elif os_auto_update_service == self.dnf_automatic:
+                    self.__init_auto_update_for_dnf_automatic()
+                    self.__get_current_auto_os_updates_setting_on_machine()
+                    if self.enable_on_reboot_value or self.apply_updates_value == 'yes':
+                        break
+                # elif os_auto_update_service == self.packagekit:
+                #     self.__init_auto_update_for_packagekit()
+                #     self.__get_current_auto_os_updates_setting_on_machine()
+                #     if self.enable_on_reboot_value or self.apply_updates_value == 'yes':
+                #         break
+
+        if self.apply_updates_value == 'no' and not self.enable_on_reboot_value:
+            return Constants.AutomaticOSPatchStates.DISABLED
+        elif self.apply_updates_value == 'yes' or self.enable_on_reboot_value:
+            return Constants.AutomaticOSPatchStates.ENABLED
+        else:
+            return Constants.AutomaticOSPatchStates.UNKNOWN
```

Comment:
Log the values that result in this outcome. Similar for why we picked enabled or disabled. Make the selection, log the selection and then the reasons for that selection, prior to returning.

Reply:
Resolved